[CORE-540] adding the context filter to the custom field listing

paultimoce opened the pull request:
The context filter on custom field listing API endpoint was released some time ago but was not yet documented.

Commit:
:memo: adding the context filter to the custom field definition listing

mrtus commented on apiary.apib:
```diff
@@ -518,6 +518,28 @@ Get a list of all the definitions of custom fields.
+                    + product
+                    + project
+                    + milestone
+                    + sale
```
B: This one should be `deal`

Some other fields are not correct either.

mrtus reviewed (changes requested):
I think we should fix our implementation first before releasing this. Half of these context fields are not correct or up to date.

/cc @stivni

stivni commented on apiary.apib:
```diff
@@ -518,6 +518,28 @@ Get a list of all the definitions of custom fields.
+                + Members
+                    + contact
+                    + company
+                    + form
```
B : this is not an existing type as far as the API is concerned
this is deprecated and we don't want it in the new API

stivni commented on apiary.apib:
```diff
@@ -518,6 +518,28 @@ Get a list of all the definitions of custom fields.
+                    + contact
+                    + company
+                    + form
+                    + event
```
B : this is not an existing type as far as the API is concerned
this is deprecated and we don't want it in the new API

stivni commented on apiary.apib:
```diff
@@ -518,6 +518,28 @@ Get a list of all the definitions of custom fields.
+                    + company
+                    + form
+                    + event
+                    + product
```
PP : this is not an existing type as far as the API is concerned
we still need to add endpoints for products, so it doesn't make sense to have the custom fields for it

stivni commented on apiary.apib:
```diff
@@ -518,6 +518,28 @@ Get a list of all the definitions of custom fields.
+                    + project
+                    + milestone
+                    + sale
+                    + project_thirdparty
```
B : this is not an existing type as far as the API is concerned
we are deprecating third parties in this form

stivni commented on apiary.apib:
```diff
@@ -518,6 +518,28 @@ Get a list of all the definitions of custom fields.
+                    + sale
+                    + project_thirdparty
+                    + invoice
+                    + inv_invoice_item
```
B : do we even want custom fields on invoice items?

stivni commented on apiary.apib:
```diff
@@ -518,6 +518,28 @@ Get a list of all the definitions of custom fields.
+                    + invoice
+                    + inv_invoice_item
+                    + subscription
+                    + meeting_report
```
B : is not part of the API yet

Commit:
:memo: cut down some fields

Only document the filtering of the contexts that are documented already in the list endpoint

paultimoce commented:
As discussed with @stivni, I added filtering documentation for the contexts that are already documented in the listing API endpoint

Will follow-up with a PR which will fix the implementation

mrtus reviewed (changes requested):
Good changes, but blocking until the implementation is fixed.

marijnvandevoorde commented:
@mrtus can you unblock? Tnx!

WouterSioen commented:
@mrnnn this should not be unblocked yet, since the implementation is not fixed yet. (We did a TA though about it and will fix this quite soon)

thomasnoe07 commented:
@dietrichm I see that this PR is still open although it is mentioned it will be fixed soon. Can you have a look if we can push this PR or not? tx

stivni commented:
👉 poking @dietrichm & @WouterSioen to see if we can still do something with this...

dietrichm commented:
This is going to be picked up within a new story in our backlog. 💪
